docs/add-your-app.md

```diff
@@ -0,0 +1,138 @@
+# Adding Your Apps
+
+
+## Automatic discovery
+While we would eventually like to automatically discover your application topology, 
+at this stage adding support for a new application is a manual step.
+The main challenge in automatic discovery stems from the fact that both kubernetes deployment descriptors is focused 
+on deployment instructions and lacks metadata required by ocopea in order to create application copies.
+
+## App templates 
+In the Ocopea terminology, every application topology is described by an "App template" entity.  
+In order to create a new app template users should use the "create-app-template" command of the hub-webapp 
+Ocopea service. See [invoking Commands](invoking-ocopea-commands.md)
+
+## Create App Template Command
+
+The "create-app-template" command expects the user to provide full description of the application topology as described
+in the 
+[hub-web swagger API](https://github.com/ocopea/orcs/blob/master/hub/webapp/web-api/src/main/resources/swagger.yaml).
+
+We'll use the lets-chat app template to understand the building block
+
+```
+{
+  "name":"lets chat",
+  "version":"1.0",
+  "description":"Lets chat cool app",
+  "entryPointServiceName":"lets-chat",
+  "appServiceTemplates":[
+    {
+      "appServiceName":"lets-chat",
+      "psbType":"k8s",
+      "imageName":"sdelements/lets-chat",
+      "imageType":"nodejs",
+      "imageVersion":"latest",
+      "psbSettings":{},
+      "environmentVariables":{
+        "LCB_DATABASE_URI":"mongodb://${chat-db.server}/letschat"
+      },
+      "dependencies":[
+        {
+          "type":"DATABASE",
+          "name":"chat-db",
+          "description":"Chats database",
+          "protocols":[
+            {
+              "protocolName":"mongodb",
+              "version":null,
+              "conditions":null
+            }
+          ]
+        }
+      ],
+      "exposedPorts":[
+        8080
+      ],
+      "httpPort":8080,
+      "entryPointUrl":"login"
+    }
+  ]
+}
+```
+
+General template fields are pretty straight forward:
```

Comment:
Hmm I thought you said you were not going to describe the API fields here as the API has it all. What changed your mind?